File: invoice-service/src/main/java/com/sudo248/invoiceservice/Controller/dto/SupplierProductDto.java
package com.sudo248.invoiceservice.Controller.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class SupplierProductDto implements Serializable {
    private SupplierDto supplier;
    private ProductDto product;
    private double distance;
    private int amountLeft;
    private double price, soldAmount, rate;
}

File: client/Soc/app/src/main/java/com/sudo248/soc/data/dto/discovery/SupplierProductDto.kt
package com.sudo248.soc.data.dto.discovery

import com.sudo248.soc.domain.entity.discovery.Route

data class SupplierProductDto(
    val supplierId: String,
    val productId: String,
    val route: Route,
    val amountLeft: Int,
    val price: Double,
    val soldAmount: Int,
    val rate: Double,
)